plugins/win-dshow/CMakeLists.txt

```diff
@@ -1,215 +1,48 @@
-if(NOT EXISTS "${CMAKE_CURRENT_SOURCE_DIR}/libdshowcapture/dshowcapture.hpp")
-  obs_status(FATAL_ERROR "libdshowcapture submodule not found! Please fetch submodules. win-dshow plugin disabled.")
-  return()
-endif()
+cmake_minimum_required(VERSION 3.24...3.25)
 
-option(ENABLE_VIRTUALCAM "Enable building with Virtual Camera (Windows)" ON)
+if(OBS_CMAKE_VERSION VERSION_GREATER_EQUAL 3.0.0)
+  add_library(obs-virtualcam-interface INTERFACE)
+  add_library(OBS::virtualcam-interface ALIAS obs-virtualcam-interface)
 
-if(NOT ENABLE_VIRTUALCAM)
-  obs_status(DISABLED "Windows Virtual Camera")
-endif()
+  target_sources(obs-virtualcam-interface INTERFACE shared-memory-queue.c shared-memory-queue.h tiny-nv12-scale.c
+                                                    tiny-nv12-scale.h)
+  target_include_directories(obs-virtualcam-interface INTERFACE "${CMAKE_CURRENT_SOURCE_DIR}")
 
-if(ENABLE_VIRTUALCAM AND NOT VIRTUALCAM_GUID)
-  set(VIRTUALCAM_GUID
-      ""
-      CACHE STRING "Virtual Camera GUID" FORCE)
-  mark_as_advanced(VIRTUALCAM_GUID)
+  include(cmake/libdshowcapture.cmake)
+  add_subdirectory(virtualcam-module)
 endif()
 
-project(win-dshow)
+legacy_check()
 
-find_package(FFmpeg REQUIRED COMPONENTS avcodec avutil)
+find_package(FFmpeg COMPONENTS avcodec avutil)
```

Comment:
Using `COMPONENTS` implicitly enables `REQUIRED`.

Reply:
I believe that's the other way around.
> A package-specific list of required components may be listed after the COMPONENTS keyword. If any of these components are not able to be satisfied, the package overall is considered to be not found. If the REQUIRED option is also present, this is treated as a fatal error, otherwise execution still continues. As a form of shorthand, if the REQUIRED option is present, the COMPONENTS keyword can be omitted and the required components can be listed directly after REQUIRED.

`COMPONENTS` does not imply `REQUIRED`. If a component is not found, the package overall will be "not found", but it will not result in a fatal error.

Reply:
Yeah my bad, it's the other way around.